tests/integration_test.rs

```diff
@@ -0,0 +1,56 @@
+/// All tests in this folder assume a server running on localhost:4242
+
+use futures::SinkExt;
+use tokio::net::TcpStream;
+use tokio::stream::StreamExt;
+use tokio_util::codec::{Framed, LengthDelimitedCodec};
+
+use bytes::Bytes;
+
+static LOCAL_REMITS: &str = "localhost:4242";
+
+macro_rules! should_respond_with {
+    ($framer:expr, $bytes:expr, $resp:expr) => {
+        $framer
```

Comment:
This looks good. Can you expand on this framer object? I see it all over but not 100% sure what its for

Reply:
Oh totally!

TCP is just bytes in, bytes out. But there isn't even a "call/response" aspect built in other than acknowledging receipts of a packet.
So if you're looping over a socket like this:

```go
message := []byte
for b := range socket {
  message = append(message, b)
}
```

How do you know when you've collected all the bytes in a message?
it can't be "when you don't get anymore" because it's possible the next byte just hasn't arrived over the wire yet.
So there are different things you can do to create "Frames" which are kinda like an http "request".
One common way is delimiters.  When \r\n comes through you know it's the end.
But that requires weird escaping rules.
What we use is called Length Delimited Framing.
Where the first x bytes (in our case 4) corresponds to the length in bytes of the rest of the message.
So you read the first 4 bytes and turn it into an int.
if you get 16, then you keep looping over the socket until you have 16 bytes, then you stop. Now you have your message! Anything else that comes over is the start of the next message.